packages/pds/src/db/migrations/20230301T003452036Z-blob-creator.ts

```diff
@@ -0,0 +1,76 @@
+import { Kysely } from 'kysely'
+import { Dialect } from '..'
+
+export async function up(db: Kysely<any>): Promise<void> {
+  await db.schema
+    .createTable('blob_new')
+    .addColumn('creator', 'varchar', (col) => col.notNull())
+    .addColumn('cid', 'varchar', (col) => col.notNull())
+    .addColumn('mimeType', 'varchar', (col) => col.notNull())
+    .addColumn('size', 'integer', (col) => col.notNull())
+    .addColumn('tempKey', 'varchar')
+    .addColumn('width', 'integer')
+    .addColumn('height', 'integer')
+    .addColumn('createdAt', 'varchar', (col) => col.notNull())
+    .addPrimaryKeyConstraint('blob_creator_pkey', ['creator', 'cid'])
+    .execute()
+
+  await db
+    .insertInto('blob_new')
+    .columns([
+      'creator',
+      'cid',
+      'mimeType',
+      'size',
+      'tempKey',
+      'width',
+      'height',
+      'createdAt',
+    ])
+    .expression((exp) =>
+      exp
+        .selectFrom('blob')
+        .innerJoin('repo_blob', 'repo_blob.cid', 'blob.cid')
```

Comment:
One potential issue that could happen here is that the same blob is in multiple commits for the same user.  I think that would lead to a constraint violate on the primary key.  It's possible just adding an on conflict clause would patch that up though.

Reply:
that's a good point 👍 
I think just the on conflict will do. I think it's reasonable to say "a given blob can only be used one way per repo"